fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java

```diff
@@ -191,16 +198,52 @@ public void dropBackend(long backendId) throws DdlException {
             throw new DdlException("Backend[" + backendId + "] does not exist");
         }
 
-        dropBackend(backend.getHost(), backend.getHeartbeatPort());
+        dropBackend(backend.getHost(), backend.getHeartbeatPort(), false);
+    }
+
+    private void checkUnforce(Backend droppedBackend) {
+        Catalog catalog = Catalog.getCurrentCatalog();
+        List<Long> tabletIds = Catalog.getCurrentInvertedIndex().getTabletIdsByBackendId(droppedBackend.getId());
+        List<Long> dbs = catalog.getDbIds();
+
+        dbs.stream().map(catalog::getDb).forEach(db ->
```

Comment:
add db read lock

Reply:
@gengjun-git done, I will remind it later.